index.js

```diff
@@ -71,7 +71,11 @@ async function startProcessAndCollectTraceData (args) {
   args.name = args.name || 'flamegraph'
 
   switch (args.kernelTracing ? platform : 'v8') {
-    case 'v8': return v8(args)
+    case 'v8': return v8(args).catch((err) => {
+      // On error, delete any V8 isolate log left over by the failed process
+      fs.unlinkSync(join(args.workingDir, v8.getIsolateLog(args.workingDir, args.pid)))
+      throw err
+    })
```

Comment:
the rationale for having these files left there is to allow debugging of 0x itself. If we have access to a `debug: true` option, we should not remove them in this case.

Reply:
Hmm, we don't really have a general --debug equivalent in 0x, and maybe we should leave the isolate logs behind on fatal errors like this by default, so people have the log first time and don't need to replicate their error with a --debug flag.

How would you feel about having this cleanup-after-error behaviour occur on the `--silent` flag, then we can use that option when deliberately causing fatal errors in CI, and probably still match user expectations?

And if that flag isn't present, maybe output a message in the console telling the user where the isolate log is and that they can use a flag to not leave create logs on errors.

Reply:
I just had another look into some internals and there are two debug options: one to debug the tree and that is very different thing and second a verbose 0x output. The latter is useful for debugging 0x itself and after giving it another thought there seems to be little benefit of keeping the isolates around in an error case. In almost all cases it won't have anything to do with the specific isolate file and if it does, it should also be easily reproducible.

Reply:
So would you be happy to delete the isolate log on fatal error by default, and keep it only if the verbose flag is used?

Reply:
Yes, that would be my favorite way of doing it.